src/Bundle/Doctrine/ORM/ExpressionBuilder.php

```diff
@@ -210,8 +222,9 @@ private function resolveFieldByAddingJoins(string $field): string
             );
             $rootAndAssociationField = sprintf('%s.%s', $rootField, $associationField);
 
-            /** @var Join[] $joins */
-            $joins = array_merge([], ...array_values($this->queryBuilder->getDQLPart('join')));
+            /** @var array<Join[]> $joinDQLPart */
```

Comment:
Shouldn't it be like so?
```suggestion
            /** @var array<Join> $joinDQLPart */
```

Reply:
This is not what the code of the Doctrine QueryBuilder et telling us (L1492 and then L1483):

This array is composed of Joins group by aliases:
```php
[
    'alias1' => Join[],
    'alias2' => Join[],
    ...
];
```